lib/SILOptimizer/Transforms/AllocBoxToStack.cpp

```diff
@@ -739,6 +746,14 @@ static bool rewriteAllocBoxAsAllocStack(AllocBoxInst *ABI) {
       continue;
     }
 
+    if (auto *oldMDI = dyn_cast<MarkDependenceInst>(User)) {
+      auto *newMDI = SILBuilderWithScope(oldMDI).createMarkDependence(
+          oldMDI->getLoc(), oldMDI->getValue(), StackBox,
```

Comment:
Maybe need to replace the value instead of the mark_dependence base. And would be worth checking the replacement in the unit test.